src/core/components/core-legend.tsx

```diff
@@ -37,10 +43,13 @@ export function ChartLegend({
   return (
     <ChartLegendComponent
       ariaLabel={ariaLabel}
-      legendTitle={title}
+      type={type}
+      title={title}
+      oppositeTitle={oppositeLegendTitle}
       items={legendItems}
       actions={actions}
       position={position}
+      bottomMaxHeight={bottomMaxHeight}
```

Comment:
Why do we implement the legend layout inside the legend itself? Instead of passing the bottomMaxHeight down to the legend, you can give extra slot in the chart-container, so that it can render either a single or a double legend.